client/app/src/main/java/com/example/criengine/Activities/CameraActivity.java

```diff
@@ -0,0 +1,176 @@
+            startActivity(intent);
+            return;
+        }
+
+        // Set the view objects.
+        deleteButton = findViewById(R.id.delete_picture);
+        cancelButton = findViewById(R.id.cancel_picture);
+        saveButton = findViewById(R.id.save_camera_image);
+        newImage = findViewById(R.id.camera_image);
+        confirmDialog = confirmDelete();
+
+        // If the book already has an image, then use that. Otherwise set a default image.
+        if (photo != null) {
+            newImage.setImageBitmap(photo);
+            deleteButton.setVisibility(View.VISIBLE);
+        } else {
+            // Only show the delete button if there is an image.
+            newImage.setImageDrawable(getResources().getDrawable(R.drawable.ic_menu_book));
+            deleteButton.setVisibility(View.GONE);
+        }
+
+        // Define the actions to perform when clicking the delete button.
+        deleteButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                book.setImageURL(null);
+                confirmDialog.show();
+            }
+        });
+
+        // Define the actions to perform when clicking the cancel button.
+        cancelButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                goBack();
+            }
+        });
+
+        // Open the camera when clicking on the image.
+        newImage.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Intent camera_intent = new Intent(MediaStore.ACTION_IMAGE_CAPTURE);
+                startActivityForResult(camera_intent, pic_id);
+            }
+        });
+
+        // Define the actions to perform when clicking the save button.
+        saveButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                if (photo != null) {
+                    dbw.uploadBookImage(book, photo);
+                }
+                goBack();
+            }
+        });
+    }
+
+    /**
+     * Retrieved from:
+     * https://stackoverflow.com/questions/11740311/android-confirmation-message-for-delete
+     * @return The confirmation dialog. If user selects to delete the book, then redirect to
+     *          the main activity.
+     */
+    private AlertDialog confirmDelete() {
+        return new AlertDialog.Builder(this)
+                // set title and message and button behaviors
+                .setTitle("Delete Book")
+                .setMessage("Are you sure you want to delete this image?")
+                .setPositiveButton("DELETE", new DialogInterface.OnClickListener() {
+                    public void onClick(DialogInterface dialog, int whichButton) {
+                        // TODO: Image for this book has already been set to null. Push changes to db.
```

Comment:
@SamLojpur This is the final part missing for this PR (database wise). We can wait for you before merging this in